pkg/application/registryConfigUseCase.go

```diff
@@ -0,0 +1,247 @@
+		}
+	}
+
+	errs = append(errs, callIfNotEmpty(ctx, entriesToSet, useCase.globalConfigRepository.SaveAll))
+	errs = append(errs, callIfNotEmpty(ctx, keysToDelete, useCase.globalConfigRepository.DeleteAllByKeys))
+
+	return errors.Join(errs...)
+}
+
+func (useCase *EcosystemRegistryUseCase) applyDoguConfigDiffs(ctx context.Context, doguName common.SimpleDoguName, diffs domain.DoguConfigDiffs) error {
+	var errs []error
+	var entriesToSet []*ecosystem.DoguConfigEntry
+	var keysToDelete []common.DoguConfigKey
+
+	for _, diff := range diffs {
+		switch diff.NeededAction {
+		case domain.ConfigActionSet:
+			entry := &ecosystem.DoguConfigEntry{
+				Key:   common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key},
+				Value: common.DoguConfigValue(diff.Expected.Value),
+			}
+			entriesToSet = append(entriesToSet, entry)
+		case domain.ConfigActionRemove:
+			keysToDelete = append(keysToDelete, common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key})
+		case domain.ConfigActionNone:
+			continue
+		default:
+			errs = append(errs, doguUnknownConfigActionError(diff.NeededAction, diff.Key.Key, doguName))
+		}
+	}
+
+	errs = append(errs, callIfNotEmpty(ctx, entriesToSet, useCase.doguConfigRepository.SaveAll))
+	errs = append(errs, callIfNotEmpty(ctx, keysToDelete, useCase.doguConfigRepository.DeleteAllByKeys))
+
+	return errors.Join(errs...)
+}
+
+func (useCase *EcosystemRegistryUseCase) applySensitiveDoguConfigDiffs(ctx context.Context, doguName common.SimpleDoguName, diffs domain.SensitiveDoguConfigDiffs) error {
+	var errs []error
+
+	var entriesToSet []*ecosystem.SensitiveDoguConfigEntry
+	var keysToDelete []common.SensitiveDoguConfigKey
+
+	for _, diff := range diffs {
+		switch diff.NeededAction {
+		case domain.ConfigActionSet:
+			entry := &ecosystem.SensitiveDoguConfigEntry{
+				Key:   common.SensitiveDoguConfigKey{DoguConfigKey: common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key}},
+				Value: common.EncryptedDoguConfigValue(diff.Expected.Value),
+			}
+			// TODO: Check if dogu is installed
+			entriesToSet = append(entriesToSet, entry)
+		case domain.ConfigActionRemove:
+			keysToDelete = append(keysToDelete, common.SensitiveDoguConfigKey{DoguConfigKey: common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key}})
+		case domain.ConfigActionNone:
+			continue
+		default:
+			errs = append(errs, doguUnknownConfigActionError(diff.NeededAction, diff.Key.Key, doguName))
+		}
+	}
+
+	errs = append(errs, callIfNotEmpty(ctx, entriesToSet, useCase.doguSensitiveConfigRepository.SaveAll))
+	errs = append(errs, callIfNotEmpty(ctx, keysToDelete, useCase.doguSensitiveConfigRepository.DeleteAllByKeys))
+
+	return errors.Join(errs...)
+}
+
+func callIfNotEmpty[T ecosystem.RegistryConfigEntry | common.RegistryConfigKey](ctx context.Context, collection []T, fn func(context.Context, []T) error) error {
+	if collection != nil && len(collection) > 0 {
```

Comment:
🚫 **[checkstyle]** <gosimple> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>S1009: should omit nil check; len() for []T is defined as zero